src/ProductConstructionService/ProductConstructionService.FeedCleaner/FeedCleaner.cs

```diff
@@ -140,39 +140,42 @@ private async Task<HashSet<string>> UpdateReleasedVersionsForPackageAsync(
             await _context.SaveChangesAsync();
         }
 
-        return releasedVersions;
+        return releasedAssets;
     }
 
     /// <summary>
     /// Deletes a version of a package from an Azure DevOps feed
     /// </summary>
     /// <param name="feed">Feed to delete the package from</param>
     /// <param name="packageName">package to delete</param>
-    /// <param name="versionsToDelete">Collection of versions to delete</param>
+    /// <param name="assetsToDelete">Collection of versions to delete</param>
     private async Task DeletePackageVersionsFromFeedAsync(
         AzureDevOpsFeed feed,
-        string packageName,
-        HashSet<string> versionsToDelete)
+        HashSet<Asset> assetsToDelete)
     {
-        foreach (string version in versionsToDelete)
+        foreach (Asset asset in assetsToDelete)
         {
             try
             {
                 _logger.LogInformation("Deleting package {package}.{version} from feed {feed}",
-                    packageName, version, feed.Name);
+                    asset.Name, asset.Version, feed.Name);
 
                 await _azureDevOpsClient.DeleteNuGetPackageVersionFromFeedAsync(
                     feed.Account,
                     feed.Project?.Name,
                     feed.Name,
-                    packageName,
-                    version);
+                    asset.Name,
+                    asset.Version);
+
+                asset.Locations.Remove(asset.Locations.First(l => l.Location.Contains(feed.Name, StringComparison.OrdinalIgnoreCase)));
```

Comment:
I'd be careful here. It might be that a package was manually uploaded to a feed and it won't be in the locations? So this will blow up? Maybe try/catch this?